movie-blog/src/components/components/contained/Item.js

```diff
@@ -0,0 +1,55 @@
+import React, {Component, useState, useEffect} from 'react';
+import Button from "../../elements/Button";
+
+const Item = ({slug, needle, genre}) => {
+  // movieDetails object is undefined upon initialization,
```

Comment:
You're supposed to have your data available all at once (within single array). This way, you may filter and sort that in more convenient way.